hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/ServerCall.java

```diff
@@ -141,14 +147,43 @@ public void done() {
     cleanup();
   }
 
+  private void release(int mask) {
+    for (;;) {
```

Comment:
Should rather wait for notifications from cleanup()?

Reply:
Sorry, I can not get your point. What's the problem here?

Reply:
I meant to put a _wait_ call inside this _for_ block, so that it does not keep iterating indefinitely. But I guess condition on #153 will already halt it on the 2nd iteration, so should be fine.

Reply:
This is for CAS. You can see the code in AtomicInteger.getAndUpdate, almost the same.